arch/riscv/core/stacktrace.c

```diff
@@ -148,8 +151,14 @@ static void walk_stackframe(stack_trace_callback_fn cb, void *cookie, const stru
 		last_fp = fp;
 		/* Unwind to the previous frame */
 		frame = (struct stackframe *)fp - 1;
-		ra = frame->ra;
-		fp = frame->fp;
+		if (esf && (esf->mepc == ra) && vrfy(frame->ra, thread, esf)) {
```

Comment:
@npitre I'm trying to discern the first frame that has a `ra` here versus without, but I only managed to get it to work sometimes with this implementation, do you have any idea on how to do it properly?